framework/src/main/java/org/checkerframework/framework/type/AnnotatedTypeFactory.java

```diff
@@ -4240,6 +4291,203 @@ private void makeGroundTargetType(
+                    && uncapturedTypeArg.getKind() == TypeKind.WILDCARD) {
+                // The type argument is a captured type. Use the type argument from the newly
+                // created and yet-to-be annotated captureType. (The annotations are added as part
+                // of capturing the wildcard.)
+                typeVarToTypeArgument.put(typeVarTypeMirror, capturedTypeArg);
+                // Also, add a mapping from the captured type mirror to the annotated captured
+                // type, so that if one captured type refers to another, the correct annotated type
+                // is used.
+                typeMirrorToCapture.put(
+                        ((AnnotatedTypeVariable) capturedTypeArg).getUnderlyingType(),
+                        capturedTypeArg);
+            } else {
+                // The type argument is not a captured type. Use the type argument from
+                // typeToCapture, which is fully-annotated.
+                typeVarToTypeArgument.put(typeVarTypeMirror, uncapturedTypeArg);
+            }
+        }
+
+        // Use the mapping above to substitute the type variables in capturedType.
+        capturedType =
+                (AnnotatedDeclaredType)
+                        typeVarSubstitutor.substituteWithoutCopyingTypeArguments(
+                                typeVarToTypeArgument, capturedType);
+
+        // Loop through the type arguments and set the annotations of each captured type.
+        List<AnnotatedTypeMirror> newTypeArgs = new ArrayList<>(typeToCapture.typeArgs);
+        List<TypeVariable> orderToCapture = TypesUtils.order(typeMirrorToCapture.keySet(), types);
+        for (TypeVariable capture : orderToCapture) {
+            AnnotatedTypeMirror capturedTypeArg = typeMirrorToCapture.get(capture);
+            int i = capturedTypeMirror.getTypeArguments().indexOf(capture);
+            AnnotatedTypeMirror uncapturedTypeArg = typeToCapture.getTypeArguments().get(i);
+            AnnotatedTypeVariable typeVariable =
+                    (AnnotatedTypeVariable) typeDeclaration.getTypeArguments().get(i);
+            captureWildcard(
+                    typeVarToTypeArgument,
+                    typeMirrorToCapture,
+                    (AnnotatedWildcardType) uncapturedTypeArg,
+                    typeVariable,
+                    (AnnotatedTypeVariable) capturedTypeArg);
+            newTypeArgs.remove(i);
+            newTypeArgs.add(i, capturedTypeArg);
+        }
+        capturedType.setTypeArguments(newTypeArgs);
+
+        capturedType.addAnnotations(typeToCapture.getAnnotations());
+        return capturedType;
+    }
+
+    /**
+     * Set the annotated bounds for fresh type variable {@code capturedArg}, so that it is the
+     * capture of {@code wildcard}.
+     *
+     * @param argMapping mapping from type variable to its type argument
+     * @param captureToAnnotatedCapture mapping from captured type mirror to captured annotated type
```

Comment:
The map's key type is TypeVariable which is a subclass of TypeMirror.  Can you be more specific than "type mirror"?

Reply:
It says "captured type mirror", not just "type mirror".